### PR TITLE
Fix Unnest.equals() by comparing field withOrdinality

### PR DESCRIPTION
**Pull request by caithagoras**

### CODE REVIEW

File: presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java [hunk 1]
```diff
@@ -156,7 +156,12 @@ protected Void visitExpression(Expression node, Integer indent)
         @Override
         protected Void visitUnnest(Unnest node, Integer indent)
         {
-            builder.append(node.toString());
+            builder.append("UNNEST(")
+                    .append(node.getExpressions().stream().map(expression->formatExpression(expression, parameters)).collect(joining(", ")))
```
Comment (electrum) [inline presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java]: Wrap stream and fix formatting

**Comment by martint** [comment]
This needs a better commit message. "Fix unnest" in the commit title is not specific enough. The title should more closely describe what this commit is about and the details should describe the motivation (i.e., what's broken) or give additional context in case the it's hard to reverse-engineer from looking at the code.

**Comment by martint** [comment]
Also, these are two independent and unrelated changes, so put them in separate commits.

File: presto-parser/src/main/java/com/facebook/presto/sql/tree/Unnest.java [hunk 2]
```diff
@@ -94,6 +94,6 @@ public boolean equals(Object obj)
             return false;
         }
         Unnest other = (Unnest) obj;
-        return Objects.equals(this.expressions, other.expressions);
+        return Objects.equals(expressions, other.expressions) && withOrdinality == other.withOrdinality;
```
Comment (findepi) [inline presto-parser/src/main/java/com/facebook/presto/sql/tree/Unnest.java]: Should the `withOrdinality` be considered in hash code too?

Comment (electrum) [inline presto-parser/src/main/java/com/facebook/presto/sql/tree/Unnest.java]: Yes, they have to agree

**Pull request merged**